docker-compose.yml

```
@@ -0,0 +1,233 @@
+          fi
+        fi
+      '
+
+  google_app_engine:
+    build: 
+      context: .
+      dockerfile: Dockerfile.google_cloud_sdk
+    container_name: oppia-app-engine
+    ports:
+      - "8181:8181"
+      - "8000:8000"
+    volumes:
+      - shared_root_dir:/
+    environment:
+      - CLOUDSDK_CORE_DISABLE_PROMPTS=1
+      - oppia_is_dockerized=true
+    depends_on:
+      webserver:
+        condition: service_healthy
+      webpack-compiler:
+        condition: service_healthy
+      datastore:
+        condition: service_healthy
+      firebase:
+        condition: service_healthy
+      redis:  
+        condition: service_healthy
+      elasticsearch:  
+        condition: service_healthy
+    healthcheck:
+      test: curl -f http://localhost:8181/ || exit 1
+      timeout: 10s
+      interval: 30s
+      retries: 5
+      start_period: 20m
+    command: >
+      sh -c '
+        if [ "$prod_env" = "true" ]; then
+          /google-cloud-sdk/bin/dev_appserver.py /app/oppia/app.yaml --runtime python38 --host=0.0.0.0 --port=8181 --admin_host=0.0.0.0 --admin_port=8000 --skip_sdk_update_check=True
+        else
+          /google-cloud-sdk/bin/dev_appserver.py /app/oppia/app_dev.yaml --runtime python38 --host=0.0.0.0 --port=8181 --admin_host=0.0.0.0 --admin_port=8000 --skip_sdk_update_check=True
+        fi
+      '
+
+  datastore:
+    build:
+      context: .
+      dockerfile: Dockerfile.google_cloud_sdk
+    container_name: oppia-cloud-datastore
+    ports:
+      - "8089:8089"
+    volumes:
+      - shared_root_dir:/
+    environment:
+      - DATASTORE_DATASET=dev-project-id
+      - DATASTORE_EMULATOR_HOST=0.0.0.0:8089
+      - DATASTORE_EMULATOR_HOST_PATH=0.0.0.0:8089/datastore
+      - DATASTORE_HOST=http://0.0.0.0:8089
+      - DATASTORE_PROJECT_ID=dev-project-id
+      - DATASTORE_USE_PROJECT_ID_AS_APP_ID=true
+      - GOOGLE_CLOUD_PROJECT=dev-project-id
+      - CLOUDSDK_CORE_DISABLE_PROMPTS=1
+      - oppia_is_dockerized=true
+    depends_on:
+      webserver:
+        condition: service_healthy
+      webpack-compiler:
+        condition: service_healthy
+    healthcheck:
+      test: curl -f http://localhost:8089/ || exit 1
+      timeout: 10s
+      interval: 2m
+      retries: 5
+      start_period: 20m
+    # Line 110 is checking if previous datastore cache exists and deleting it if it does as user has not passed the save_datastore flag.
+    # TODO: thouroughly test this.
+    command: >
+      sh -c '
+        if [ "$save_datastore" = "true" ]
+        then
+          /google-cloud-sdk/bin/gcloud beta emulators datastore start --project dev-project-id --data-dir /app/cloud_datastore_emulator_cache --host-port 0.0.0.0:8089 --consistency=1.0 --quiet
+        else
+          if [ -d /app/cloud_datastore_emulator_cache ]
+          then
+            rm -rf /app/cloud_datastore_emulator_cache
+          fi
```

```suggestion
          rm -rf /app/cloud_datastore_emulator_cache || true
```
remove if exist else true instead of throwing error.